src/restconfig/src/main/java/org/geoserver/rest/catalog/DataStoreFileController.java

```diff
@@ -492,6 +519,14 @@ public void dataStorePut(
                     if (!uploadedFile.parent().delete()) {
                         LOGGER.info("Unable to delete " + uploadedFile.path());
                     }
+                } else if (uploadedFile.getType() == Resource.Type.DIRECTORY) {
+                    for (Resource file : files) {
```

Comment:
Resource.delete removes files recursively when called, it should be enough. You can check the return value to see if the contents have been fully removed, or not. Or you did experience single file locking issues? In any case, a warning is issued by Files.emtpyDirectory if that happens.

Reply:
Hi @aaime well the problem here is a bit different. Basically "uploadedFile" for some type of DataStore it is the folder containing all the uploaded files. "files" contains the list of the files uploaded on the current session and the flag "canDelete" tells me if I need to remove them or not.

I have chosen this approach in order to delete just the uploaded files selectively instead of cleaning the whole parent folder. Do you think that this is correct?

Reply:
Ah ok, sorry, misread the code :-)